### PR TITLE
Board with drag and drop

### LINKED ISSUE #69: Board
Implement the dynamic board, which will be used to play. We want it to set up dynamically and use the custom size provided. Check if its possible to interact with grid elements dynamically, if not probably we can set it up with individual fields.

TODO:
- Analyze if its feasible to use the grid Control to set up the board (check if it would work properly for the drag and drop purposes)
- Initialize the grid with tiles from #67 after its complete
- Set up basic movement mechanics

### PR DESCRIPTION
**Pull request by Fruktus**
Closes #69

### CODE REVIEW

File: QuadradiusR/scripts/board.gd [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		var player2_torus = torus_template.instance().init(self)
+		
+		board.get_child(i).set_slot(player1_torus)
+		board.get_child(board_size * board_size - 1 - i).set_slot(player2_torus)
+
+
+# After picking up torus, move it to top of the tree, so it won't get covered
+# by other sprites. After putting down, place back on original position, unless
+# it was a new tile, in this case move to it
+func _torus_pickup(torus):
```
Comment (Observer46) [inline QuadradiusR/scripts/board.gd:46]: add typehinting pls

File: QuadradiusR/scripts/board.gd [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	add_child(torus)
+
+
+func _torus_putdown():
+	print('putting down')
+	remove_child(self.active_torus)
+	torus_parent.add_child(active_torus)
+
+
+func _on_tile_release(active_tile):
```
Comment (Observer46) [inline QuadradiusR/scripts/board.gd:60]: t y p e h i n t i n g

File: QuadradiusR/scripts/board.gd [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+func _torus_pickup(torus):
+	self.active_torus = torus
+	self.torus_parent = torus.get_parent()
+	print('from', torus_parent.get_parent().position)  # DEBUG
+	torus.get_parent().remove_child(torus)
+	add_child(torus)
+
+
+func _torus_putdown():
+	print('putting down')
```
Comment (Observer46) [inline QuadradiusR/scripts/board.gd:55]: add `# DEBUG` comment

File: QuadradiusR/scripts/board.gd [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	remove_child(self.active_torus)
+	torus_parent.add_child(active_torus)
+
+
+func _on_tile_release(active_tile):
+	print('putting down new tile')
```
Comment (Observer46) [inline QuadradiusR/scripts/board.gd:61]: add `# DEBUG` comment

File: QuadradiusR/scripts/tile.gd [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+var position: Vector2
+
+
+
+func init(position: Vector2):
+	self.position = position
+	return self
+
+
+func set_slot(torus: Control):
```
Comment (Observer46) [inline QuadradiusR/scripts/tile.gd:17]: POG CHAMP

Comment (Observer46) [reply QuadradiusR/scripts/tile.gd:17]: can add!

File: QuadradiusR/scripts/torus.gd [hunk 6]
```diff
@@ -3,31 +3,62 @@ extends Control
...
 	set_process(false)
 
 
-func _process(delta):
-	self.rect_global_position = get_global_mouse_position() - rect_size/2
+func _process(delta: float):
+	self.rect_global_position = get_global_mouse_position() - rect_size/2 * board.rect_scale
+
+
+func init(board):
```
Comment (Observer46) [inline QuadradiusR/scripts/torus.gd:25]: t       y          p             e          h  i n        t      in g

File: QuadradiusR/scripts/torus.gd [hunk 7]
```diff
@@ -3,31 +3,62 @@ extends Control
...
+	set_process(false)
+	get_tree().call_group("torus", "set_interaction", true)
+	get_tree().call_group("board", "_torus_putdown")
+	is_held = false
+	# TODO play putdown sfx, but always or only on tile change?
+	self.rect_scale = Vector2(1, 1)
+	rect_position = self.starting_pos
+
+
+func set_interaction(can_interact):
```
Comment (Observer46) [inline QuadradiusR/scripts/torus.gd:58]: typehinting

File: QuadradiusR/scripts/tile.gd [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+	
+	slot.add_child(torus)
+
+
+func _on_mouse_event(viewport: Node, event: InputEvent, shape_idx: int):
+	if event is InputEventMouseButton:
+		print(event)
+		if event.get_button_index() == BUTTON_LEFT and not event.is_pressed():
+			print('active', position)
+			get_tree().call_group("board", "_on_tile_release", self)
```
Comment (Observer46) [inline QuadradiusR/scripts/tile.gd:29]: event overload - needs a rework

**Review by Observer46 — APPROVED** [review]
As a base version - it's oke!

File: QuadradiusR/scripts/tile.gd (hunk 8 above)
Comment (Fruktus) [reply QuadradiusR/scripts/tile.gd:29]: Moved to issue #93

**Pull request merged**